contracts/src/utils/structs/checkpoints.rs

```diff
@@ -29,6 +32,12 @@ pub enum Error {
     CheckpointUnorderedInsertion(CheckpointUnorderedInsertion),
 }
 
+impl MethodError for Error {
```

Comment:
Can we have #[derive(MethodError)]` macro?

Reply:
Also don't like it. Basically I have proposed it in [pr to stylus-sdk](https://github.com/OffchainLabs/stylus-sdk-rs/pull/136). I hope they will do smth with it:)

Reply:
Also we don't have a proc macro library for contracts. Do you think we should create it for this temporary macro?

Reply:
Instead of separate crate for `derive` macro. I've added functional macro that looks like this: `impl_method_error!(Error);`. Wdt @bidzyyys?

Reply:
decided with @alexfertel to write it expicitly